modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_zephyr.c

```diff
@@ -201,3 +256,101 @@ uint64_t nrf_802154_platform_sl_lptimer_sync_lpticks_get(void)
 {
 	return m_sync_timer.target_time;
 }
+
+nrf_802154_sl_lptimer_platform_result_t nrf_802154_platform_sl_lptimer_hw_task_prepare(
+	uint64_t fire_lpticks,
+	uint32_t ppi_channel)
+{
+	bool done_on_time;
+	nrf_802154_sl_mcu_critical_state_t mcu_cs_state;
+
+	if (!hw_task_state_set(HW_TASK_STATE_IDLE, HW_TASK_STATE_SETTING_UP)) {
+		/* The only one available set of peripherals is already used. */
+		return NRF_802154_SL_LPTIMER_PLATFORM_NO_RESOURCES;
+	}
+
+	/* Allocate the channel if not already done. */
+	if (m_hw_task.chan == RTC_CHAN_INVALID) {
+		/* The channel allocation cannot take place during module
+		 * initialization, because for nRF53 it would run
+		 * out of resources - some other module temporarily needs
+		 * an rtc channel to initialize. For this reason, this is where
+		 * the allocation is made.
+		 * Once a channel has been successfully allocated, it will be held
+		 * until nrf_802154_platform_sl_lp_timer_deinit is called.
+		 */
+		m_hw_task.chan = z_nrf_rtc_timer_chan_alloc();
```

Comment:
Is this function called from any timing critical code, like RADIO IRQ handlers?

Reply:
It is called from the RTC IRQ handler

Reply:
In such case I think it would be better to handle sharing resources in the correct initialization sequence, instead of trying to permanently allocate resources run-time.

Reply:
The sync_rtc module allocate the cc until the offset establishing algorithm is completed, which takes several hundred RTC ticks from launching NET core. This extends beyond the SYS_INIT based initialization sequence.
Therefore, I do not think that anything can be improved by changing the order of initialization of modules, if that's what you mean.

Reply:
Ok, I wasn't aware of that. Probably adding more features to the allocator (queues?) would be the proper solution, but the gain is probably not worth the cost of implementation.